code/datums/action.dm

```diff
@@ -708,19 +726,60 @@
 	button.maptext_height = 12
 	return button
 
+/datum/action/cooldown/Destroy()
+	if(!isnull(sequence_actions))
+		sequence_actions.Cut()
+	QDEL_LIST(initialized_actions)
+	return ..()
+
+/datum/action/cooldown/Grant(mob/M)
+	. = ..()
+	if(!owner)
+		return
+	UpdateButtons()
+	if(next_use_time > world.time)
+		START_PROCESSING(SSfastprocess, src)
+	RegisterSignal(owner, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, .proc/handle_melee_attack)
+	for(var/datum/action/cooldown/ability as anything in initialized_actions)
+		ability.Grant(owner)
+
+/datum/action/cooldown/Remove(mob/M)
```

Comment:
Same here - changing `M` to something like `removed_from` is appreciated

(Only need to do it with this override, don't need to do it it for all `Remove()`s)